cedar-policy/src/api/err.rs

```diff
@@ -517,45 +517,122 @@ impl From<cedar_policy_validator::ValidationWarning> for ValidationWarning {
     }
 }
 
+/// Error structs for the variants of `PolicySetError`
+pub mod policy_set_error_structs {
+    use super::Error;
+    use crate::PolicyId;
+    use miette::Diagnostic;
+
+    /// There was a duplicate [`PolicyId`] encountered in either the set of
+    /// templates or the set of policies.
+    #[derive(Debug, Diagnostic, PartialEq, Eq, Error, Clone)]
+    #[error("duplicate template or policy id `{id}`")]
+    pub struct AlreadyDefined {
+        pub(crate) id: PolicyId,
+    }
+
+    /// Expected a static policy, but a template-linked policy was provided
+    #[derive(Debug, Diagnostic, PartialEq, Eq, Error, Clone)]
+    #[error("expected a static policy, but a template-linked policy was provided")]
+    pub struct ExpectedStatic {}
```

Comment:
Not sure if we want empty structs for these two. I did it to be consistent, but can be persuaded otherwise.

Reply:
I think they're nice. Makes it easier to add more detail to the error message in the future for pretty much no cost now.

Reply:
I'm convinced by @john-h-kastner-aws as well. We should add structs for empty variants.